File: URI_Judge/1068.py
while True:
    try:
        expressao = input()
        abre = 0
        fecha = 0

        fechouAntesDeAbrir = False

        for letra in expressao:
            if letra == '(':
                abre += 1
            if letra == ')':
                fecha += 1
            if fecha > abre:
                fechouAntesDeAbrir = True
        if fecha == abre and not fechouAntesDeAbrir:
            print('correct')
        else:
            print('incorrect') 
    except EOFError:
        break

File: URI_Judge/1068.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  string word;
  int abre, fecha;
  while(getline(cin, word)){
    abre = 0, fecha = 0;
    for(char letter: word) {
      if(letter == '(')
        abre++;
      if(letter == ')'){
        if(fecha + 1 > abre){
          fecha = -1;
          break;
        }
        fecha++;
      }
    }

    if(abre == fecha) {
      cout << "correct" << endl;
    }else {
      cout << "incorrect" << endl;
    }
  }

  return 0;
}